stonesoup/metricgenerator/tests/test_uncertaintymetric.py

```diff
@@ -0,0 +1,92 @@
+"""Uncertainty metric tests."""
+import datetime
+
+import numpy as np
+
+from ..manager import SimpleManager
+from ..uncertaintymetric import UncertaintyMetric
+from ...types.detection import Detection
+from ...types.groundtruth import GroundTruthPath, GroundTruthState
+from ...types.state import State, GaussianState
+from ...types.track import Track
+
+
+def test_uncertaintymetric_extractstates():
+    """Test uncertainty metric extract states."""
+    generator = UncertaintyMetric()
+
+    # Test state extraction
+    time_start = datetime.datetime.now()
+    detections = [Detection(state_vector=np.array([[i]]), timestamp=time_start)
+                  for i in range(5)]
+    tracks = {Track(states=[State(state_vector=[[i]],
+                                  timestamp=time_start)]) for i in range(5)}
+    truths = {GroundTruthPath(states=[GroundTruthState(state_vector=[[i]],
+                                                       timestamp=time_start)])
+              for i in range(5)}
+
+    det_states = generator.extract_states(detections)
+    assert det_states.states == detections
+    track_states = generator.extract_states(tracks)
+    assert set(track_states) == set(t.states[0] for t in tracks)
+    truth_states = generator.extract_states(truths)
+    assert set(truth_states) == set(t.states[0] for t in truths)
+
```

Comment:
Adding a test to make sure error is raised when you extract states of something non-`state` and non-`statemutablesequence` will increase code coverage.